src/cache/inmemory/inMemoryCache.ts

```diff
@@ -226,7 +227,7 @@ export class InMemoryCache extends ApolloCache<NormalizedCacheObject> {
     }
   }
 
-  public diff<TData, TVariables = any>(
+  public diff<TData, TVariables extends OperationVariables = any>(
```

Comment:
This could possibly be changed to

```suggestion
  public diff<TData, TVariables extends OperationVariables = OperationVariables>(
```